artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -319,8 +327,13 @@ protected void ensureScanning() {
     * @return a copy of the {@link DeliveryAnnotations} present in the message or null if non present.
     */
    public final DeliveryAnnotations getDeliveryAnnotations() {
-      ensureScanning();
-      return scanForMessageSection(deliveryAnnotationsPosition, DeliveryAnnotations.class);
+
+      if (deliveryAnnotations != null) {
+         return deliveryAnnotations;
```

Comment:
I would break contract and fix the java doc here, as we were not using deliveryAnnotations before.

if you have a strong opinion I can create a new method.

Reply:
it is actually a copy.. it's just parsed during decoding.

I tried to keep the old semantics however I need this on large messages (as I would have to open the large message file lazily just to decode this, what would suck).

Reply:
It is parsed at decoding, but its not a copy at the time of the call which was the intent of the method contracts...meaning now the actual value can be manipulated outside the message contract. For example allowing things to be added/used in a non thread safe manner...or that then get discarded if the message data is scanned again and this new field that was added gets overrwriten..or etc.

Reply:
Ok, fair enough.. I will set this to null as soon as the method is called, further calls would be the previous semantic.

I need to decode the annotation during the initial scan, as I won't be able to do this lazily for large messages.

Reply:
I can't really do that though.. as if an user highjack the call with an annotation would have the broker use case broken.

I will create a new method.. I don't think it's a big deal, but if you prefer it this way I will do it.

Reply:
that won't make things clear though, as we now have two methods.. one for the initial scan, and one that return the copy... there's no point really.